crates/system-manager-engine/src/main.rs

```diff
@@ -0,0 +1,134 @@
+//! System Manager Engine
+//!
+//! This binary handles privileged operations for system-manager:
+//! - activate: Apply /etc files and start systemd services
+//! - deactivate: Remove managed configuration
+//! - prepopulate: Place files without starting services
+//! - register: Register a store path as the active profile
+//!
+//! It is designed to be invoked by the system-manager CLI wrapper,
+//! either directly (with sudo) or via SSH for remote deployments.
+//! This allows uniform handling of privilege escalation.
+
+use anyhow::Result;
+use clap::Parser;
+use std::process::ExitCode;
+
+use system_manager_engine::{NixOptions, StorePath, PROFILE_DIR};
+
+#[derive(clap::Parser, Debug)]
+#[command(
+    author,
+    version,
+    about = "System Manager Engine - privileged operations for system-manager"
+)]
+struct Args {
+    #[command(subcommand)]
+    action: Action,
+
+    #[clap(long = "nix-option", num_args = 2, global = true)]
+    nix_options: Option<Vec<String>>,
+}
+
+#[derive(clap::Args, Debug)]
+struct ActivationArgs {
+    #[arg(long, action)]
+    /// If true, only write under /run, otherwise write under /etc
+    ephemeral: bool,
+}
+
+#[derive(clap::Args, Debug)]
+struct StorePathArg {
+    #[arg(long = "store-path")]
+    /// The store path containing the system-manager profile
```

Comment:
The doc comment should go above the attribute, but also the `long = "store-path"` can be reduced to just `long` here.